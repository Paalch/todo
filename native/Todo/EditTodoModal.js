import React, { Component } from 'react'
import { View, Label, Item, Input, Text, Content, Title ,Icon , Button, Form } from 'native-base';
import { Modal } from 'react-native'

export default class EditTodoModal extends Component<{}>  {

    constructor(props) {
        super(props)

        this.state = {
            tempContent: this.props.content,
            content: this.props.content,
            date: this.props.date
        }
    }

    handleButtonSaveClick = () => {
        const { tempContent } = this.state
        if(tempContent !== "") {
            this.props.onButtonSaveClick(tempContent)
            this.setState({
                tempContent: '',
                content: ''
            })

            this.props.toggleModal()
        }
    }

    handleButtonEditClick = () => {
        const { handleDelete } = this.props
        handleDelete()
        this.handleButtonSaveClick()
    }

    render() {
        const { toggleModal, isOpen, onButtonSaveClick , content} = this.props
        const { tempContent } = this.state
        return (
            <Modal
                animationType="slide"
                transparent={false}
                visible={isOpen}
                onRequestClose={() => {toggleModal()}}
            >

                <View style={{flex:1}}>
                    <Content >
                        <Form>
                            <View style={{flex:1}}>
                                <Item floatingLabel>
                                    <Label>
                                        Todo
                                    </Label>
                                    <Input
                                        onChangeText={(tempContent) => this.setState({tempContent})}
                                        value={tempContent}
                                    />
                                </Item>
                            </View>
                        </Form>
                    </Content>

                </View>
                <View style={{position:'absolute',bottom:0, width:'100%'}}>
                    <Button block success onPress={this.handleButtonEditClick}>
                        <Text>Save</Text>
                    </Button>

                </View>
            </Modal>
        )
    }
}