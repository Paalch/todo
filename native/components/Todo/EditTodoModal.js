import React, { Component } from 'react'
import { View, Item, Input, Text, Content, Title, Button, Form, H1, Grid, Col } from 'native-base';
import { Modal, TouchableHighlight } from 'react-native'
import DefaultHeader from '../DefaultHeader'
import { modalButtons, views } from '../../styles'


export default class EditTodoModal extends Component  {

    constructor(props) {
        super(props)

        //Sets state, tempText is the text in the textfield and text is the text from the todo object
        this.state = {
            tempText: this.props.text,
            text: this.props.text,
        }
    }

    /**
     * Button handler for save button
     */
    handleButtonSaveClick = () => {
        const { tempText } = this.state
        //Checks that the text in not empty
        if(tempText !== "") {
            this.props.onButtonSaveClick(tempText)
            //resets the state
            this.setState({
                tempText: '',
                text: ''
            })
            //Hides the modal
            this.props.toggleModal()
            //removes the old todo
            this.props.handleDelete()
        }
    }

    /**
     * Handle for button close click.
     * Resets the original text to state
     * Closes model
     */
    handleButtonCloseClick = () => {
        const {toggleModal} = this.props
        toggleModal()

    }

    render() {
        //define constants
        const { toggleModal, isOpen } = this.props
        const { tempText } = this.state

        return (
            <Modal
                animationType="slide"
                transparent={false}
                visible={isOpen}
                onRequestClose={() => {toggleModal()}}>
                <DefaultHeader title={"Edit Todo"} toggleModal={toggleModal}/>
                <View style={views.flex1}>
                    <Content>
                        <Form>
                            <View style={views.flex1}>
                                <H1>Write your todo here:</H1>
                                <Item>
                                    <Input
                                        onChangeText={(tempText) => this.setState({tempText})}
                                        value={tempText}
                                        placeholde="Write your todo here"/>
                                </Item>
                            </View>
                        </Form>
                    </Content>
                </View>

                <View style={modalButtons.view}>
                    <Grid>
                        <Col>
                            <TouchableHighlight
                                onPress={this.handleButtonSaveClick}
                                style={modalButtons.save}>

                                <Text style={modalButtons.text}>
                                    SAVE
                                </Text>

                            </TouchableHighlight>
                        </Col>
                        <Col>

                            <TouchableHighlight
                                onPress={this.handleButtonCloseClick}
                                style={modalButtons.close}>

                                <Text style={modalButtons.text}>
                                    CANCEL
                                </Text>
                            </TouchableHighlight>

                        </Col>
                    </Grid>
                </View>

            </Modal>

        )
    }
}
