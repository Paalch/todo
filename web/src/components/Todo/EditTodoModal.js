import React, { Component } from 'react'
import { Button, Modal, Grid, Form, TextArea } from 'semantic-ui-react'

// Define style for grid
const dateStyle = {
    fontSize: '1rem',
    color: '#999999'
}

export default class EditToDoModal extends Component {
    constructor (props) {
        super(props)

        // Init state
        this.state = {
            tempContent: this.props.content,
            content: this.props.content,
            date: this.props.date
        }
    }

    // Monitor tempContent input
    handleContentInput = (e, {value}) => {
        this.setState({
            tempContent: value
        })
    }

    // Saves the changes made, on save button click
    handleButtonSaveClick = () => {
        const { tempContent } = this.state
        if (tempContent !== '') {
            this.props.onButtonSaveClick(tempContent)
            this.setState({
                tempContent: '',
                text: ''
            })

            // Close the window
            this.props.onClose()
            // delete old todo
            this.props.handleDelete()
        }
    }

    // closes the window, no change
    handleButtonClose = () => {
        const { content } = this.state
        this.setState({
            tempContent: content
        })
        this.props.onClose()
    }

    render () {
        // define constants
        const { isOpen } = this.props
        const { tempContent, date } = this.state

        // checks if the window is open or not
        if (!this.props.isOpen) {
            return null
        }
        return (
            <Modal
                open={isOpen}
                size='tiny'>
                <Modal.Header>
                    <Grid width={16} >
                        <Grid.Column width={10}>
                            Edit your todo:
                        </Grid.Column>
                        <Grid.Column
                            width={6}
                            textAlign="right"
                            verticalAlign="middle"
                            style={dateStyle}>
                            { date }
                        </Grid.Column>
                    </Grid>
                </Modal.Header>
                <Modal.Content scrolling>

                    <Modal.Description>
                        <Form>
                          <TextArea
                              onChange={ this.handleContentInput }
                              value={ tempContent } size='small'
                              placeholder='ToDo text ...'
                              autoHeight={true}/>
                        </Form>
                    </Modal.Description>

                </Modal.Content>
                <Modal.Actions>
                    <Button.Group>
                        <Button
                            onClick={ this.handleButtonSaveClick }
                            color='green'>
                            Save
                        </Button>
                        <Button
                            onClick={this.handleButtonClose}
                            color='grey'>
                            Close
                        </Button>
                    </Button.Group>
                </Modal.Actions>
            </Modal>
        )
    }
}
