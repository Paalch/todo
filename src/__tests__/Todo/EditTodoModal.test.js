import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import EditTodoModal from '../../components/Todo/EditTodoModal';
import Adapter from 'enzyme-adapter-react-15'
import toJson from 'enzyme-to-json'
import moment from 'moment'


Enzyme.configure({ adapter: new Adapter() });

test('EditTodoModal component should render as expected when isOpen is true', () => {

  const component = shallow(<EditTodoModal
    content={"You should do this thing right now"}
    date={moment('20171018')}
    isOpen={true}/>);

  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});

test('EditTodoModal component should not render at all when isOpen is false', () => {
  const component = shallow(<EditTodoModal isOpen={false}/>);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});