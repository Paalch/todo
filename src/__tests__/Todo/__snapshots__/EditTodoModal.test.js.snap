// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditTodoModal component should not render at all when isOpen is false 1`] = `""`;

exports[`EditTodoModal component should render as expected when isOpen is true 1`] = `
<Modal
  closeOnDimmerClick={true}
  closeOnDocumentClick={false}
  dimmer={true}
  eventPool="Modal"
  open={true}
  size="tiny"
>
  <ModalHeader>
    <Grid
      width={16}
    >
      <GridColumn
        width={10}
      >
        Edit your todo:
      </GridColumn>
      <GridColumn
        style={
          Object {
            "color": "#999999",
            "fontSize": "1rem",
          }
        }
        textAlign="right"
        verticalAlign="middle"
        width={6}
      />
    </Grid>
  </ModalHeader>
  <ModalContent
    scrolling={true}
  >
    <ModalDescription>
      <Form
        as="form"
      >
        <TextArea
          as="textarea"
          autoHeight={true}
          onChange={[Function]}
          placeholder="ToDo text ..."
          rows={3}
          size="small"
          value="You should do this thing right now"
        />
      </Form>
    </ModalDescription>
  </ModalContent>
  <ModalActions>
    <ButtonGroup>
      <Button
        as="button"
        color="green"
        onClick={[Function]}
      >
        Save
      </Button>
      <Button
        as="button"
        color="grey"
        onClick={[Function]}
      >
        Close
      </Button>
    </ButtonGroup>
  </ModalActions>
</Modal>
`;
